interpreter/utils/count_tokens.py

```diff
@@ -0,0 +1,38 @@
+import tiktoken
+from litellm import cost_per_token
+
+def count_tokens(text="", model="gpt-4"):
+    """
+    Count the number of tokens in a string
+    """
+
+    encoder = tiktoken.encoding_for_model(model)
+
+    return len(encoder.encode(text))
+
+def token_cost(tokens=0, model="gpt-4"):
+    """
+    Calculate the cost of the current number of tokens
+    """
+
+    (prompt_cost, _) = cost_per_token(model=model, prompt_tokens=tokens)
+
+    return prompt_cost
+
+def count_messages_tokens(messages=[], model=None):
+    """
+    Count the number of tokens in a list of messages
+    """
+
+    tokens_used = 0
+
+    for message in messages:
+        if isinstance(message, str):
+            tokens_used += count_tokens(message, model=model)
+        elif "message" in message:
+            tokens_used += count_tokens(message["message"], model=model)
```

Comment:
Upon further investigation it seems like this approach might miss the generated code from Open Interpreter when calculating token usage.

I need to tweak things to take that into account as well.

Reply:
I think this is more accurate now with the recent push.